src/nvim/ops.c

```diff
@@ -2524,6 +2528,58 @@ static void yank_copy_line(yankreg_T *reg, struct block_def *bd, long y_idx)
   *pnew = NUL;
 }
 
+static void yank_do_autocmd(oparg_T *oap, yankreg_T *reg)
+{
+  char buf[NUMBUFLEN+2];
+
+  if (!has_event(EVENT_TEXTYANKPOST)) {
+    // No autocommand was defined
+    return;
+  }
+
+  // set v:event to a dictionary with information about the yank
+  dict_T *dict = dict_alloc();
+
+  // the yanked text
+  list_T *list = list_alloc();
+  for (int i = 0; i < reg->y_size; i++) {
+    list_append_string(list, reg->y_array[i], -1);
+  }
+  dict_add_list(dict, "regcontents", list);
+
+  // the register type
+  buf[0] = NUL;
+  buf[1] = NUL;
+  switch (reg->y_type) {
+  case MLINE: buf[0] = 'V'; break;
+  case MCHAR: buf[0] = 'v'; break;
+  case MBLOCK:
+    buf[0] = Ctrl_V;
+    snprintf(buf + 1, ARRAY_SIZE(buf) - 1, "%" PRId64,
+             (int64_t)(reg->y_width + 1));
+    break;
+  default:
+    assert(false);
```

Comment:
AFAIR compiler does not bother warning about no `default` with standard types. At least, both clang and gcc have nothing against

```C
int main(int argc, char **argv, char **environ)
{
    switch (argc) {
        case 1: break;
    }
    return 0;
}
```

(actually, the same code, but without line breaks because I used `echo … | {compiler} -xc {-Wall(gcc)|-Weverything(clang)} -`; clang though complained about unused arguments).